src/controllers/Auth.ts

```diff
@@ -0,0 +1,529 @@
+import { Jwt } from '../core/Jwt';
+import { BaseController } from './Base';
+import { User } from '../core/security/User';
+import { JSONObject, ApiKey } from '../utils/interfaces';
+
+/**
+ * Auth controller
+ *
+ * @param kuzzle
+ * @constructor
+ */
+export class AuthController extends BaseController {
+  private _authenticationToken: any | null;
+
+  /**
+   * constructor
+   * @param kuzzle
+   */
+  constructor (kuzzle) {
+    super(kuzzle, 'auth');
+
+    this._authenticationToken = null;
+
+    this.kuzzle.on('tokenExpired', () => {
+      this._authenticationToken = null;
+    });
+  }
+
+  /**
+   *  Authentication token in use
+   */
+  get authenticationToken (): any | null {
+    return this._authenticationToken;
+  }
+
+  set authenticationToken (encodedJwt: any) {
+    if (encodedJwt === undefined || encodedJwt === null) {
+      this._authenticationToken = null;
+    }
+    else if (typeof encodedJwt === 'string') {
+      this._authenticationToken = new Jwt(encodedJwt);
+    }
+    else {
+      throw new Error(`Invalid token argument: ${encodedJwt}`);
+    }
+  }
+
+  /**
+   * Do not add the token for the checkToken route, to avoid getting a token error when
+   * a developer simply wish to verify his token
```

Comment:
Use the gender-neutral [singular they](https://en.wikipedia.org/wiki/Singular_they):

```suggestion
   * a developer simply wishes to verify their token
```